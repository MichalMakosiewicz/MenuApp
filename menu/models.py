from django.db import models
import os
from uuid import uuid4
from django.utils.deconstruct import deconstructible


@deconstructible
class PathAndRename(object):

    def __init__(self, sub_path):
        self.path = sub_path

    def __call__(self, instance, filename):
        ext = filename.split('.')[-1]
        filename = '{}.{}'.format(uuid4().hex, ext)
        return os.path.join(self.path, filename)


class Dish(models.Model):
    name = models.CharField(max_length=50)
    description = models.CharField(max_length=300)
    price = models.IntegerField()
    preparation_time = models.IntegerField()
    add_date = models.DateField()
    update_date = models.DateField()
    photo = models.ImageField(upload_to=PathAndRename("static"), default="Empty")
    is_vegan = models.BooleanField()

    def __str__(self):
        return self.name


class MenuCard(models.Model):
    title = models.CharField(max_length=50)
    create_date = models.DateField()
    update_date = models.DateField()
    dish = models.ManyToManyField(Dish)

    def __str__(self):
        return self.title
