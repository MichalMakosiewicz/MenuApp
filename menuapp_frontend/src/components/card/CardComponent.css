.root {
    display: flex;
    justify-content: flex-start;
}

.MuiCardContent-root{
    display: flex;
    flex-direction: column;
    align-self: flex-start;
}

.title {
    font-size: 23px;
}

.pos {
    margin-bottom: 12px;
}
