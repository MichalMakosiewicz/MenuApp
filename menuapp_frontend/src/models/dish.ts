export interface DishInterfaceRaw {
    id: number;
    name: string;
    description: string;
    price: number;
    preparation_time: number;
    add_date: string;
    update_date: string;
    photo: string;
    is_vegan: boolean;
}

export class DishModel {
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public preparation_time: number;
    public add_date: string;
    public update_date: string;
    public photo: string;
    public is_vegan: boolean;

    constructor(dishInterface: DishInterfaceRaw) {
        this.id = dishInterface.id;
        this.name = dishInterface.name;
        this.description = dishInterface.description;
        this.price = dishInterface.price;
        this.preparation_time = dishInterface.preparation_time;
        this.add_date = dishInterface.add_date;
        this.update_date = dishInterface.update_date;
        this.photo = dishInterface.photo;
        this.is_vegan = dishInterface.is_vegan;
    }
}