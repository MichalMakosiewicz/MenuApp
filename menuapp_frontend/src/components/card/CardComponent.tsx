import React, {useEffect} from 'react';
import {DishModel} from '../../models/dish';
import {CardModel} from '../../models/card';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import {DishComponent} from '../dish/DishComponent';
import "./CardComponent.css"

interface CardPropsInterface {
    card: any;
    dishes: DishModel[];
}

export function CardComponent(props: CardPropsInterface) {
    const {card, dishes} = props;

    const cardDishes = card.dish.map((dish: number, index: string | number | null | undefined) => {
        if (!card.dish) {
            return [];
        }
        // @ts-ignore
        return <DishComponent key={index} dish={dishes.find(model => model.id === dish)}/>
    });

    return (
        <div>
            <Card className='root'>
                <CardContent>
                    <Typography className="title" variant="h5" gutterBottom>
                        {card.title}
                    </Typography>
                    <div className="info">
                        <Typography color="primary" variant="subtitle2">
                            last update date: {card.update_date}
                        </Typography>
                        <Typography color="primary" variant="subtitle2">
                            creation date: {card.create_date}
                        </Typography>
                    </div>
                    <br/>
                    <div className="dish-boundary">
                        {cardDishes}
                    </div>
                </CardContent>
                <br />
            </Card>
        </div>
    )
}