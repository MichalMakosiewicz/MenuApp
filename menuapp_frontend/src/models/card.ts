import { DishModel } from './dish';

export interface CardInterfaceRaw {
    id: number;
    title: string;
    create_date: string;
    update_date: string;
    dish: number[];
}

export class CardModel {
    public id: number;
    public title: string;
    public create_date: string;
    public update_date: string;
    public dish: number[];

    constructor(cardInterface: CardInterfaceRaw) {
        this.id = cardInterface.id;
        this.title = cardInterface.title;
        this.create_date = cardInterface.create_date;
        this.update_date = cardInterface.update_date;
        this.dish = cardInterface.dish;
    }
}