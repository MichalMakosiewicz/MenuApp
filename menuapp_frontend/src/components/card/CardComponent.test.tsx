import React from 'react';
import {CardComponent} from './CardComponent';
import renderer from 'react-test-renderer';
import {CardModel} from '../../models/card';

describe('Navbar test', () => {
    const card = new CardModel({
        id: 1,
        title: "test",
        create_date: "test",
        update_date: "test",
        dish: []
    })
    it('match the snapshot', () => {
        const tree = renderer.create(<CardComponent card={card} dishes={[]}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})