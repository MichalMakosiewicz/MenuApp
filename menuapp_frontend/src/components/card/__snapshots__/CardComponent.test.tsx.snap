// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test match the snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <h5
        className="MuiTypography-root title MuiTypography-h5 MuiTypography-gutterBottom"
      >
        test
      </h5>
      <div
        className="info"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
        >
          last update date: 
          test
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
        >
          creation date: 
          test
        </h6>
      </div>
      <br />
      <div
        className="dish-boundary"
      />
    </div>
    <br />
  </div>
</div>
`;
